src/bin/triplydb-create-query-job.ts

```diff
@@ -107,29 +89,40 @@ const command = program
     const account = await app.getUser(options.account);
     // check whether account name exists
     await account.getInfo();
-    const queries: { queryId: string; priority?: number }[] = [];
-    for (const query of queryInfo) {
-      const queryAccount = await app.getAccount(query.queryAccountName);
-      const queryId = (await (await queryAccount.getQuery(query.queryName)).getInfo()).id;
-      queries.push({ queryId, priority: query.priority });
-    }
-    const sourceDatasetAccount = await app.getAccount(sourceDatasetAccountName);
-    const sourceDatasetId = (await (await sourceDatasetAccount.getDataset(sourceDatasetName)).getInfo()).id;
-    const targetDatasetAccount = await app.getAccount(targetDatasetAccountName);
-    const targetDatasetId = (await (await targetDatasetAccount.ensureDataset(targetDatasetName)).getInfo()).id;
-    const payload: QueryJobPipelineCreate = {
-      queries: queries,
-      sourceDatasetId: sourceDatasetId,
-      targetDatasetId: targetDatasetId,
-      targetGraphName: options.targetGraphName,
-    };
-    const queryJob: QueryJob = new QueryJob(app, account);
-    try {
-      await queryJob.createQueryJobPipeline(payload, queryInfo);
-    } catch (e) {
-      console.error(e);
-      process.exit(1);
-    }
+    readJson(configFile)
+      .then(async (data) => {
+        const queryJobConfig: QueryJobPipelineCreate = data;
+        if (!queryJobConfig) {
+          sanityCheckError("Error in reading query job json config");
+        }
+        if (!("version" in queryJobConfig)) {
+          sanityCheckError("Version not found in query job config");
+        }
+
+        const queryInfo: QueryInformation[] = [];
+        if ("queries" in queryJobConfig && queryJobConfig.queries) {
+          for (const query of queryJobConfig.queries as any[]) {
+            const [queryAccountName, actualQueryName] = query.name.split("/");
+            if (!queryAccountName) sanityCheckError(`Missing query account name for query "${query.name}"`);
+            if (!actualQueryName) sanityCheckError(`Missing query name for query "${query.name}"`);
+            queryInfo.push({
+              queryAccountName: queryAccountName,
+              queryName: actualQueryName,
+              priority: query.priority || 0,
+            });
+          }
+        }
+        const queryJob: QueryJob = new QueryJob(app, account);
+        try {
+          await queryJob.createQueryJobPipeline(queryJobConfig, queryInfo);
+        } catch (e) {
+          console.error(e);
+          process.exit(1);
+        }
+      })
+      .catch((e) => {
+        if (e) sanityCheckError(e.message);
```

Comment:
Why call sanitycheckerror? If I understand correctly (based on the name), this is for handling sanity checks of the CLI arguments, showing e.g. the help text. Do we expect to catch such errors in this clause?

In addition to that: we're doing a truthy check on `e`. What situation do we cater for by doing that?